synapse/rest/client/v2_alpha/sync.py

```diff
@@ -229,7 +234,12 @@ async def encode_response(self, time_now, sync_result, access_token_id, filter):
                 "left": list(sync_result.device_lists.left),
             },
             "presence": SyncRestServlet.encode_presence(sync_result.presence, time_now),
-            "rooms": {"join": joined, "invite": invited, "leave": archived},
+            "rooms": {
+                "join": joined,
+                "invite": invited,
+                "knock": knocked,
```

Comment:
I think this is supposed to use an unstable key?

Reply:
Yep, membership events and join rules need to use the unstable prefix as well, whee!